src/qt/listctrl.cpp

```diff
@@ -1507,6 +1507,11 @@ wxRect wxListCtrl::GetViewRect() const
     return rect;
 }
 
+wxSize wxListCtrl::DoGetBestClientSize() const
+{
+    return wxQtConvertSize(m_qtTreeWidget->minimumSizeHint());
+}
+
```

Comment:
Thanks for fixing this, but I think it would be better to call `viewportSizeHint()` here instead of using the minimum size by default. Just add this `using QTreeView::viewportSizeHint;` too to the public interface of `wxQtListTreeWidget` because it is protected in `QTreeView` and we need to call it from `wxListCtrl`.

TIA.

Reply:
Thanks for the suggestion.

If there's a lot of items, `viewportSizeHint` returns 254x868 here, for example:
![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/e8621118-77d7-4693-99fc-b5cba7129f1c)

And because best size == effective min size here, it can't be shrinked.

For reference, on wxGTK it looks like this:

![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/efd877ef-646c-4e6b-8f65-693d2da44eab)

Reply:
On wxQt with `minimumSizeHint` it looks like this:

![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/1a3749ba-3603-46c5-8567-f419bab6111a)

Reply:
I thought we should follow the other ports (I can only test under Windows & Linux) when calculating the best client size. i.e. Both `wxMSW` and `wxGTK` do this:
- the calculated best width is slightly larger than the widest string.
- the calculated best height is limited to 10 items.

AFAICT `minimumSizeHint()` always returns a fixed size, but the problem with `viewportSizeHint()` can be solved by limiting the returned height to 10 items as is done by `wxMSW` and `wxGTK`.

Please let me know if I'm still missing something here? TIA.

Reply:
When using `averageCharWidth`, in both wxGTK and wxQt, `DoGetBestClientSize` returns 400x160 size

Reply:
wxGTK:
![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/eeae2495-f8a8-49ec-9b1d-39f02b4a3fcd)

wxQt:
![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/71b2a63e-dcca-4389-8d99-f0c13dec7163)

Reply:
I don't recall why we had to use `maxWith` instead of `averageCharWidth` TBH, and I tried to run the tests locally with `maxWidth` replaced by `averageCharWidth` and everything runs successfully. So unless anyone remembers why `maxWith` is used instead of `averageCharWidth`, let's change `wxQtDCImpl::GetCharWidth` to use the latter then and see how it goes. TIA.